src/python/pants/backend/python/goals/export.py

```diff
@@ -401,6 +566,18 @@ async def export_virtualenv_for_resolve(
             editable_local_dists_digest=editable_local_dists_digest,
         ),
     )
+
+    # Add generated Python sources from codegen targets to the virtualenv.
+    if export_subsys.options.py_generated_sources:
+        if export_subsys.options.py_resolve_format == PythonResolveExportFormat.mutable_virtualenv:
+            export_result = await add_codegen_to_export_result(
+                request.resolve, export_result, codegen_setup
+            )
+        else:
+            logger.warning(
```

Comment:
I'm not sure if this should be a warning, since it doesn't represent bad user behavior. Maybe just `info`?

Also, it would be nice if the message linked to the references pages for the two relevant options.

Reply:
Will do.